Cloud SaaS team page revamp (#934)
sourcegraph/handbook: content/engineering/cloud/saas/hiring/cloud-saas-software-engineer-architectural-interview.md

# Architectural Interview

> [This interview](https://github.com/sourcegraph/interviews/blob/master/engineering/software-engineer/architecture/repo-management/README.md) only applies for engineers who apply to [Core Application](../index.md) team,

Reviewer: ```suggestion
[This interview](https://github.com/sourcegraph/interviews/blob/master/engineering/software-engineer/architecture/repo-management/README.md) only applies for engineers who apply to [Cloud SaaS](../index.md) team,
```

> The interview consists of open-ended questions, in order to get you to describe your mental model of designing a new system from scratch. We want to understand if you are making the right decisions when designing system architectures. We expect you to ask clarifying questions as the requirements for the system are on purpose ambiguously defined.

Reviewer: ```suggestion
The interview consists of open-ended questions, allowing you to describe your mental model of designing a new system from scratch - we want to understand the decisions you are making when designing system architectures. We expect you to ask clarifying questions as the requirements for the system are purposefully ambiguous.
```

Reviewer: Dropped right (as there are often no right / wrong decisions, only tradeoffs).

## What do we want to learn?

- How do you identify and resolve ambiguity?
- How do you make tradeoffs?
- How do you communicate your proposal?
- What is it like to work with you on a problem you haven’t seen before?
- What is your intuition about solutions that are likely to yield good results?

Why? Read the [FAQ](#FAQ).

## FAQ

### Why do you do an architectural interview?

> In certain roles, we expect you to be designing systems to meet often conflicting or ambiguous requirements. For engineers in these roles a significant part of their journey in Sourcegraph is going to be designing new or improving existing systems to fit these new requirements. For engineers in these roles to be successful, we are looking for skills, that are best described above in [What do we want to learn](#what-do-we-want-to-learn) section.

Reviewer: ```suggestion
In certain roles, we expect you to be designing systems that often need to meet conflicting or ambiguous requirements. For engineers in these roles a significant part of their journey in Sourcegraph is going to be designing new or improving existing systems to fit these new requirements. For engineers in these roles to be successful, we are looking for skills, that are best described above in [What do we want to learn](#what-do-we-want-to-learn) section.
```
